pkg/ignition/config.go

```diff
@@ -0,0 +1,59 @@
+package ignition
+
+import (
+	"encoding/json"
+)
+
+// Config is an ignition config.
+type Config struct {
+	// Ignition is metadata for the configuration itself.
+	Ignition Ignition `json:"ignition"`
+	// Storage describes the desired state of the system's storage devices.
+	Storage *Storage `json:"storage,omitempty"`
+	// Systemd describes the desired state of the systemd units.
+	Systemd *Systemd `json:"systemd,omitempty"`
+}
+
+// Ignition is the metadata for the configutation itself.
+type Ignition struct {
+	// Version is the semantic version number of the spec.
+	Version string `json:"version"`
+}
+
+// Storage describes the desired state of the system's storage devices.
+type Storage struct {
+	// Files is the list of files to be written.
+	Files []File `json:"files,omitempty"`
+}
+
+// WriteConfig marshals the ignition config into json.
+func (i *Config) WriteConfig() ([]byte, error) {
+	i.Ignition.Version = "2.2.0"
+	return json.MarshalIndent(i, "", "  ")
+}
+
+// AddFiles adds the specified files to the ignition config.
+func (i *Config) AddFiles(files ...File) {
+	if i.Storage == nil {
+		i.Storage = &Storage{}
+	}
+	if i.Storage.Files == nil {
```

Comment:
This can all be collapsed into the following line:

```go
  i.Storage.Files = append(i.Storage.Files, files...)
```

Reply:
Of course. That was pretty boneheaded of me.